lib/filter/tests/test_filters_fop_cmp.c

```diff
@@ -21,72 +22,234 @@
+  cr_assert_not(evaluate("10", KW_NUM_GE, "11"));
+  cr_assert_not(evaluate("10", KW_NUM_GT, "11"));
+
+  cr_assert_not(evaluate("11", KW_NUM_LT, "10"));
+  cr_assert_not(evaluate("11", KW_NUM_LE, "10"));
+  cr_assert_not(evaluate("11", KW_NUM_EQ, "10"));
+  cr_assert(evaluate("11", KW_NUM_GE, "10"));
+  cr_assert(evaluate("11", KW_NUM_GT, "10"));
+}
+
+/* string comparison */
+
+Test(filter, test_eq)
+{
+  cr_assert(evaluate("10", KW_EQ, "10"));
+  cr_assert(evaluate("$SEVERITY_NUM", KW_EQ, "7"));
+  cr_assert(evaluate("$SEVERITY_NUM", KW_EQ, "$SEVERITY_NUM"));
+
+  cr_assert_not(evaluate("10", KW_EQ, "11"));
+
+}
+
+Test(filter, test_ne)
+{
+  cr_assert(evaluate("10", KW_NE, "9"));
+  cr_assert(evaluate("$SEVERITY_NUM", KW_NE, "8"));
+  cr_assert(evaluate("$SEVERITY_NUM", KW_NE, "$FACILITY_NUM"));
+
+  cr_assert_not(evaluate("10", KW_NE, "10"));
+}
+
+Test(filter, test_lt)
+{
+  cr_assert(evaluate("10", KW_LT, "11"));
+  cr_assert(evaluate("7", KW_LT, "8"));
+  cr_assert(evaluate("$LEVEL_NUM", KW_LT, "8"));
+
+  cr_assert_not(evaluate("7", KW_LT, "10"));
+  cr_assert_not(evaluate("11", KW_LT, "10"));
+  cr_assert_not(evaluate("11", KW_LT, "11"));
+}
+
+Test(filter, test_le)
+{
+  cr_assert(evaluate("11", KW_LE, "11"));
+  cr_assert(evaluate("10", KW_LE, "11"));
+  cr_assert(evaluate("7", KW_LE, "8"));
+  cr_assert(evaluate("$LEVEL_NUM", KW_LE, "8"));
+
+  cr_assert_not(evaluate("7", KW_LE, "10"));
+  cr_assert_not(evaluate("11", KW_LE, "10"));
+  cr_assert_not(evaluate("$LEVEL_NUM", KW_LE, "10"));
+}
+
+Test(filter, test_gt)
+{
+  cr_assert(evaluate("11", KW_GT, "10"));
+  cr_assert(evaluate("8", KW_GT, "7"));
+  cr_assert(evaluate("8", KW_GT, "$LEVEL_NUM"));
+
+  cr_assert_not(evaluate("10", KW_GT, "7"));
+  cr_assert_not(evaluate("10", KW_GT, "11"));
+  cr_assert_not(evaluate("10", KW_GT, "10"));
+  cr_assert_not(evaluate("10", KW_GT, "$LEVEL_NUM"));
+}
+
+Test(filter, test_ge)
+{
+  cr_assert(evaluate("10", KW_GE, "10"));
+  cr_assert(evaluate("11", KW_GE, "10"));
+  cr_assert(evaluate("8", KW_GE, "7"));
+  cr_assert(evaluate("8", KW_GE, "$LEVEL_NUM"));
+
+  cr_assert_not(evaluate("10", KW_GE, "7"));
+  cr_assert_not(evaluate("10", KW_GE, "11"));
+  cr_assert_not(evaluate("10", KW_GE, "$LEVEL_NUM"));
+}
+
+Test(filter, test_string_ordering)
+{
+  cr_assert_not(evaluate("10", KW_LT, "10"));
+  cr_assert(evaluate("10", KW_LE, "10"));
+  cr_assert(evaluate("10", KW_EQ, "10"));
+  cr_assert(evaluate("10", KW_GE, "10"));
+  cr_assert_not(evaluate("10", KW_GT, "10"));
+
+  cr_assert(evaluate("10", KW_LT, "11"));
+  cr_assert(evaluate("10", KW_LE, "11"));
+  cr_assert_not(evaluate("10", KW_EQ, "11"));
+  cr_assert_not(evaluate("10", KW_GE, "11"));
+  cr_assert_not(evaluate("10", KW_GT, "11"));
+
+  cr_assert_not(evaluate("11", KW_LT, "10"));
+  cr_assert_not(evaluate("11", KW_LE, "10"));
+  cr_assert_not(evaluate("11", KW_EQ, "10"));
+  cr_assert(evaluate("11", KW_GE, "10"));
+  cr_assert(evaluate("11", KW_GT, "10"));
```

Comment:
I think it would also be worth adding a test case that documents the weirdness of string comparison applied on numeric characters, for example:

```
"11" is KW_GT than "1000"
```

Reply:
I think this is included, with "7" and "10" which is the same problem.

Reply:
Sorry, I did not see them.